packages/@lwc/babel-plugin-component/src/decorators/wire/transform.ts

```diff
@@ -174,9 +175,10 @@ function buildWireConfigValue(t: BabelTypes, wiredValues: WiredValue[]) {
 
             if (wiredValue.params) {
                 const dynamicParamNames = wiredValue.params.map((p) => {
-                    return t.stringLiteral(
-                        t.isIdentifier(p.key) ? p.key.name : (p.key as types.StringLiteral).value
-                    );
+                    const value = t.isIdentifier(p.key)
+                        ? p.key.name
+                        : (p.key as types.StringLiteral).value;
+                    return p.computed ? t.identifier(value) : t.stringLiteral(value);
```

Comment:
The issue here isn't really Symbols – it's the use of the computed object property:

```js
  @wire(getFoo, { [key1]: "$prop1" })
```

The `[key1]` (computed property) here is the issue. But there is actually a case that this code misses, which is a string literal inside of the computed property:

```js
  @wire(getFoo, { ["foo"]: "$prop1" })
```

This incorrectly produces the identifier `foo` rather than the string `"foo"`:

```js
      dynamic: [foo], // <-- incorrect identifier, points to nothing
      config: function ($cmp) {
        return {
          ["foo"]: $cmp.prop1
        };
      }
```

I would recommend adding a test for the above case. (Yes it's an unnecessary use of the computed property, but we should do the right thing if people are zany enough to use it. 🙂) I'd also change the test name to `computed-properties` rather than `symbol-properties`.